Ignore dotfiles when discovering magic files

Lekensteyn opened the pull request:
I was a bit surprised to get an "UnicodeDecodeError: 'utf-8' codec can't
decode byte 0xbe in position 21: invalid start byte" error while
writing and testing a new magic file.

It turns out that binwalk tried to parse the Vim swap file (.magic.swp)
as a magic file which obviously fails. Therefore ignore all files not
starting with an alphanumeric character.
___
The check could be made even stronger such that files like `magic.orig`, `magic.dpkg-old` or `magic.pacnew` are ignored, but this is already a good start I guess. Maybe the plugin path discovery needs a similar fix?

devttys0 commented:
I agree with ignoring hidden files, as that can cause a good deal of confusion. Given that many other standard Linux utilities also ignore hidden files by default (e.g., `ls`), this should not be surprising behavior to an end user. It would be nice if there were an easy cross-platform way of ignoring hidden files, but most people run binwalk in Linux, so just filtering on the first character is probably sufficient. 

I don't agree with ignoring *all* files that don't start with a letter or a number. If a user wants to name their file `_myfile`, that's their business. Likewise, if they accidentally leave files they don't want binwalk to parse laying around in a directory that binwalk parses, that's their problem. Trying to anticipate the user's intentions rarely works well.

Lekensteyn commented:
I've added a new commit with the original rationale and restricted the filter to dotfiles only. Tested by creating `~/.binwalk/magic/test` containing hex `\xff`, then ran `file README.md` (failed as expected). Renamed it to `.test` and the problem is gone.

devttys0 commented:
Thanks!
